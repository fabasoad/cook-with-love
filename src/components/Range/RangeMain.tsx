/// <reference path='./RangeMain.d.ts' />
import * as React from 'react'
import { useTranslation } from 'react-i18next'

import RangeItem from './RangeItem'

import imgArkaDeutscheLieferadresse from '../../assets/images/range/deutsche_lieferadresse_website.jpg'
import imgArkaInvebstor from '../../assets/images/range/invebstor.jpg'
import imgArkaMedisport from '../../assets/images/range/medisport_website.jpg'
import imgArkaPrintAndPost from '../../assets/images/range/print_and_post_website.jpg'
import imgArkaWebsite from '../../assets/images/range/arka_website.png'
import imgArkaWorkPlus from '../../assets/images/range/work_plus.png'
import imgArkaZakupki from '../../assets/images/range/zakupki_website.png'
import imgBitcoinComGames from '../../assets/images/range/bitcoin.com_games.png'
import imgBitcoinComGamesAdmin from '../../assets/images/range/bitcoin.com_games_admin.png'
import imgBitcoinComGamesAndroid from '../../assets/images/range/bitcoin.com_android.png'
import imgBitcoinComMarkets from '../../assets/images/range/bitcoin.com_markets.png'
import imgLohikaHpeAlm from '../../assets/images/range/alm.jpg'
import imgLohikaHpeAlmRobot from '../../assets/images/range/alm_robot.png'
import imgLohikaHpeLabManagement from '../../assets/images/range/lab_management.jpg'
import imgLohikaEarnin from '../../assets/images/range/earnin.png'
import imgLohikaPrysm from '../../assets/images/range/prysm_iphone_app.png'
import imgSMA from '../../assets/images/range/sma.png'

export default function RangeMain() {
  const { t } = useTranslation()
  return (
    <div id="range">
      <div className="container">
        <div className="section-title text-center center">
          <h2>{t('cook-with-love-range-title')}</h2>
          <hr />
        </div>
        <div className="row">
          <RangeItem
            url="https://markets.bitcoin.com/"
            name="Markets"
            title={t('cook-with-love-range-markets-title')}
            subtitle={t('cook-with-love-range-markets-subtitle')}
            img={imgBitcoinComMarkets}
          />
          <RangeItem
            url="https://games.bitcoin.com/"
            name="Games"
            title={t('cook-with-love-range-casino-web-app-title')}
            subtitle={t('cook-with-love-range-casino-web-app-subtitle')}
            img={imgBitcoinComGames}
          />
          <RangeItem
            url="https://games.bitcoin.com/"
            name="Games Admin"
            title={t('cook-with-love-range-casino-admin-web-app-title')}
            subtitle={t('cook-with-love-range-casino-admin-web-app-subtitle')}
            img={imgBitcoinComGamesAdmin}
          />
          <RangeItem
            url="https://games.bitcoin.com/android"
            name="Games Android"
            title={t('cook-with-love-range-casino-android-app-title')}
            subtitle={t('cook-with-love-range-casino-android-app-subtitle')}
            img={imgBitcoinComGamesAndroid}
          />
          <RangeItem
            url="https://www.earnin.com/"
            name="Earnin"
            title={t('cook-with-love-range-earnin-title')}
            subtitle={t('cook-with-love-range-earnin-subtitle')}
            img={imgLohikaEarnin}
          />
          <RangeItem
            url="https://www.prysm.com/software/iphone-app/"
            name="Prysm iPhone App"
            title={t('cook-with-love-range-prysm-title')}
            subtitle={t('cook-with-love-range-prysm-subtitle')}
            img={imgLohikaPrysm}
          />
          <RangeItem
            url="https://drive.google.com/drive/folders/163jhUuVrBCEYQ503sa1Y1M7KG--1gwaj?usp=sharing"
            name="Southern Marine Agency Website"
            title={t('cook-with-love-range-sma-title')}
            subtitle={t('cook-with-love-range-sma-subtitle')}
            img={imgSMA}
          />
          <RangeItem
            url="https://admhelp.microfocus.com/alm/en/12.60/online_help/Content/Admin/sa_robot_toc.htm"
            name="ALM Robot"
            title={t('cook-with-love-range-alm-robot-title')}
            subtitle={t('cook-with-love-range-alm-robot-subtitle')}
            img={imgLohikaHpeAlmRobot}
          />
          <RangeItem
            url="https://www.microfocus.com/en-us/products/application-lifecycle-management/overview"
            name="ALM"
            title={t('cook-with-love-range-alm-title')}
            subtitle={t('cook-with-love-range-alm-subtitle')}
            img={imgLohikaHpeAlm}
          />
          <RangeItem
            url="https://admhelp.microfocus.com/alm/en/12.60/online_help/Content/lm_ag.htm"
            name="Lab Management"
            title={t('cook-with-love-range-alm-lab-management-title')}
            subtitle={t('cook-with-love-range-alm-lab-management-subtitle')}
            img={imgLohikaHpeLabManagement}
          />
          <RangeItem
            url="https://youtu.be/8wTcyZEtQx8"
            name="Invebstor"
            title={t('cook-with-love-range-invebstor-title')}
            subtitle={t('cook-with-love-range-invebstor-subtitle')}
            img={imgArkaInvebstor}
          />
          <RangeItem
            url="https://youtu.be/eRxTKZ0knsg"
            name="Work+"
            title={t('cook-with-love-range-work-plus-title')}
            subtitle={t('cook-with-love-range-work-plus-subtitle')}
            img={imgArkaWorkPlus}
          />
          <RangeItem
            url="https://www.arka-media.de/"
            name="ARKA Website"
            title={t('cook-with-love-range-arka-title')}
            subtitle={t('cook-with-love-range-arka-subtitle')}
            img={imgArkaWebsite}
          />
          <RangeItem
            url="https://www.deutsche-lieferadresse.com/"
            name="Deutsche Lieferadresse Website"
            title={t('cook-with-love-range-deutsche-lieferadresse-title')}
            subtitle={t('cook-with-love-range-deutsche-lieferadresse-subtitle')}
            img={imgArkaDeutscheLieferadresse}
          />
          <RangeItem
            url="http://medisport-garmisch.de/"
            name="Medisport Website"
            title={t('cook-with-love-range-medisport-title')}
            subtitle={t('cook-with-love-range-medisport-subtitle')}
            img={imgArkaMedisport}
          />
          <RangeItem
            url="#range"
            name="Print &amp; Post Website"
            title={t('cook-with-love-range-print-and-post-title')}
            subtitle={t('cook-with-love-range-print-and-post-subtitle')}
            img={imgArkaPrintAndPost}
          />
          <RangeItem
            url="https://zakupki-de.com.ua/"
            name="Zakupki Website"
            title={t('cook-with-love-range-zakupki-title')}
            subtitle={t('cook-with-love-range-zakupki-subtitle')}
            img={imgArkaZakupki}
          />
        </div>
      </div>
    </div>
  )
}
