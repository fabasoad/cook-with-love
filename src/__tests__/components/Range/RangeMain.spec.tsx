/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import RangeMain from '../../../components/Range/RangeMain'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render RangeMain correctly', () => {
  const totalNumberOfProducts: number = 21;
  const wrapper: ShallowWrapper = shallow(<RangeMain />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(totalNumberOfProducts)
  expect(tMock).toHaveBeenNthCalledWith(1, 'cook-with-love-range-title')
  for (let i = 1; i < totalNumberOfProducts; i += 2) {
    expect(tMock).toHaveBeenNthCalledWith(
      i + 1, expect.stringMatching(/^cook-with-love-range-.*-title$/)
    )
  }
})
