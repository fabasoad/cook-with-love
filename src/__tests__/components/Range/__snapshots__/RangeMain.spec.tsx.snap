// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RangeMain correctly 1`] = `
<div
  id="range"
>
  <div
    className="container"
  >
    <div
      className="section-title text-center center"
    >
      <h2>
        cook-with-love-range-title
      </h2>
      <hr />
    </div>
    <div
      className="row"
    >
      <RangeItem
        img="test-file-stub"
        name="Markets"
        subtitle="cook-with-love-range-markets-subtitle"
        title="cook-with-love-range-markets-title"
        url="https://markets.bitcoin.com/"
      />
      <RangeItem
        img="test-file-stub"
        name="Games"
        subtitle="cook-with-love-range-casino-web-app-subtitle"
        title="cook-with-love-range-casino-web-app-title"
        url="https://games.bitcoin.com/"
      />
      <RangeItem
        img="test-file-stub"
        name="Games Admin"
        subtitle="cook-with-love-range-casino-admin-web-app-subtitle"
        title="cook-with-love-range-casino-admin-web-app-title"
        url="https://games.bitcoin.com/"
      />
      <RangeItem
        img="test-file-stub"
        name="Games Android"
        subtitle="cook-with-love-range-casino-android-app-subtitle"
        title="cook-with-love-range-casino-android-app-title"
        url="https://games.bitcoin.com/android"
      />
      <RangeItem
        img="test-file-stub"
        name="Earnin"
        subtitle="cook-with-love-range-earnin-subtitle"
        title="cook-with-love-range-earnin-title"
        url="https://www.earnin.com/"
      />
      <RangeItem
        img="test-file-stub"
        name="Prysm iPhone App"
        subtitle="cook-with-love-range-prysm-subtitle"
        title="cook-with-love-range-prysm-title"
        url="https://www.prysm.com/software/iphone-app/"
      />
      <RangeItem
        img="test-file-stub"
        name="Southern Marine Agency Website"
        subtitle="cook-with-love-range-sma-subtitle"
        title="cook-with-love-range-sma-title"
        url="https://drive.google.com/drive/folders/163jhUuVrBCEYQ503sa1Y1M7KG--1gwaj?usp=sharing"
      />
      <RangeItem
        img="test-file-stub"
        name="ALM Robot"
        subtitle="cook-with-love-range-alm-robot-subtitle"
        title="cook-with-love-range-alm-robot-title"
        url="https://admhelp.microfocus.com/alm/en/12.60/online_help/Content/Admin/sa_robot_toc.htm"
      />
      <RangeItem
        img="test-file-stub"
        name="ALM"
        subtitle="cook-with-love-range-alm-subtitle"
        title="cook-with-love-range-alm-title"
        url="https://www.microfocus.com/en-us/products/application-lifecycle-management/overview"
      />
      <RangeItem
        img="test-file-stub"
        name="Lab Management"
        subtitle="cook-with-love-range-alm-lab-management-subtitle"
        title="cook-with-love-range-alm-lab-management-title"
        url="https://admhelp.microfocus.com/alm/en/12.60/online_help/Content/lm_ag.htm"
      />
      <RangeItem
        img="test-file-stub"
        name="Invebstor"
        subtitle="cook-with-love-range-invebstor-subtitle"
        title="cook-with-love-range-invebstor-title"
        url="https://youtu.be/8wTcyZEtQx8"
      />
      <RangeItem
        img="test-file-stub"
        name="Work+"
        subtitle="cook-with-love-range-work-plus-subtitle"
        title="cook-with-love-range-work-plus-title"
        url="https://youtu.be/eRxTKZ0knsg"
      />
      <RangeItem
        img="test-file-stub"
        name="ARKA Website"
        subtitle="cook-with-love-range-arka-subtitle"
        title="cook-with-love-range-arka-title"
        url="https://www.arka-media.de/"
      />
      <RangeItem
        img="test-file-stub"
        name="Deutsche Lieferadresse Website"
        subtitle="cook-with-love-range-deutsche-lieferadresse-subtitle"
        title="cook-with-love-range-deutsche-lieferadresse-title"
        url="https://www.deutsche-lieferadresse.com/"
      />
      <RangeItem
        img="test-file-stub"
        name="Medisport Website"
        subtitle="cook-with-love-range-medisport-subtitle"
        title="cook-with-love-range-medisport-title"
        url="http://medisport-garmisch.de/"
      />
      <RangeItem
        img="test-file-stub"
        name="Print & Post Website"
        subtitle="cook-with-love-range-print-and-post-subtitle"
        title="cook-with-love-range-print-and-post-title"
        url="#range"
      />
      <RangeItem
        img="test-file-stub"
        name="Zakupki Website"
        subtitle="cook-with-love-range-zakupki-subtitle"
        title="cook-with-love-range-zakupki-title"
        url="https://zakupki-de.com.ua/"
      />
    </div>
  </div>
</div>
`;
